File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/constructor/WeekConstructor.java
package com.tickaroo.tikxml.annotationprocessing.textcontent.constructor;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Xml;
import java.util.List;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "week")
public class WeekConstructor {

  private List<DayConstructor> days;

  public WeekConstructor(
      @Element List<DayConstructor> days) {
    this.days = days;
  }

  public List<DayConstructor> getDays() {
    return days;
  }

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof WeekConstructor)) return false;

    WeekConstructor that = (WeekConstructor) o;

    return days != null ? days.equals(that.days) : that.days == null;
  }

  @Override public int hashCode() {
    return days != null ? days.hashCode() : 0;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/textcontent/constructor/WeekConstructor.kt
package com.tickaroo.tikxml.annotationprocessing.textcontent.constructor

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "week")
class WeekConstructor(
        @param:Element val days: List<DayConstructor>?) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is WeekConstructor) return false

        val that = other as WeekConstructor?

        return if (days != null) days == that!!.days else that!!.days == null
    }

    override fun hashCode(): Int {
        return days?.hashCode() ?: 0
    }
}
